core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -759,15 +780,17 @@ public ReplicationPhase(ReplicaRequest replicaRequest, Response finalResponse, S
             for (ShardRouting shard : shards) {
                 if (shard.primary() == false && executeOnReplica == false) {
                     numberOfIgnoredShardInstances++;
-                } else if (shard.unassigned()) {
+                    continue;
+                }
+                if (shard.unassigned()) {
```

Comment:
can we add comment telling people this mimics the logic in doRun and they should look there for more information